### ISSUE TITLE
Verifying an exception thrown in Future's thread

### ISSUE THREAD

**Issue by matwojcik** [etorreborre/specs2#456]
I wanted to check if the exception was thrown in the future thread - so that Future eventually is Failure(e). But I discovered, that despite using matcher with .await, it passes also exceptions thrown in the caller thread, and the test is green. Example:

```scala
class SomeTest extends Specification with specification.mutable.ExecutionEnvironment {
  def is(implicit ee: ExecutionEnv) = {
    def someFunc(condition: Boolean): Future[String] = {
      if (condition) Future {
        "someString"
      }
      else throw new UnsupportedOperationException
    }

    "someFunc" should {
      "should fail that test" in {
        someFunc(false) must throwA[UnsupportedOperationException].await
      }
    }
  }
}
```

I expect that test to fail, but it passes. If I move the result to a variable: 

```scala
"should fail that test" in {
        val result = someFunc(false)
        result must throwA[UnsupportedOperationException].await
}
```

it fails as it should.

I'd like that test to pass for function like:

```scala
    def someFunc(condition: Boolean): Future[String] = Future {
      if (condition)  {
        "someString"
      }
      else throw new UnsupportedOperationException
    }
```

It happens on 3.7. I'm not sure if the intention was to be working like that, and I need to write my own Matcher for that?

**Comment by etorreborre** [comment]
This looks like a bug, I'll come back to you.

**Comment by etorreborre** [comment]
This looks right to me. `someFunc(false)` is throwing an exception so `someFunc(false) must throwA[UnsupportedOperationException].await` should be a success.

When you put `someFunc(false)` in a variable the exception gets thrown right away and reported by specs2 as an error.

**Comment by matwojcik** [comment]
What I expected is whenever I put .await after the matcher, it is checked the future result instead of immediate result. If it was not the intention for that matcher, I would definitely like the idea of having special matcher for that in specs2.

**Comment by etorreborre** [comment]
I just realised that the 2 `someFunc` methods you wrote are different. The first one throws an exception right away if the condition is false:
```scala
def someFunc(condition: Boolean): Future[String] = {
      if (condition) Future {
        "someString"
      }
      else throw new UnsupportedOperationException
    }
```
and the second one throws an exception *inside* the Future:
```scala
def someFunc(condition: Boolean): Future[String] = Future {
      if (condition) Future {
        "someString"
      }
      else throw new UnsupportedOperationException
    }
```

Which one do you want to test?

**Comment by matwojcik** [comment]
These are two different implementation of function defined as 
```scala
def someFunc(condition: Boolean): Future[String]
```

I'd like that impl to fail the test:
```scala
def someFunc(condition: Boolean): Future[String] = {
      if (condition) Future {
        "someString"
      }
      else throw new UnsupportedOperationException
    }
```

and that to pass the test:
```scala
def someFunc(condition: Boolean): Future[String] = Future {
      if (condition)  {
        "someString"
      }
      else throw new UnsupportedOperationException
    }
```

The background for that is - I have made a couple of mistakes in the code, throwing the error in the caller thread instead of future thread. I was convinced that those kind of matchers would pass only those impl, that make future failed with exception in future thread.

**Comment by matwojcik** [comment]
Thx a lot!

**Comment by etorreborre** [comment]
Thanks for the explanation, this was indeed a bug. The fix can be found in version `3.7-20160217162713-3fed6b7` (published in a few minutes).

**Comment by etorreborre** [comment]
The new jars are accessible now, in particular [this one](https://oss.sonatype.org/content/repositories/releases/org/specs2/specs2-matcher_2.11/3.7-20160217162713-3fed6b7/) containing the fix.
